### PR TITLE
profiler: add tag to indicate whether execution tracing is enabled

### PR DESCRIPTION
**Pull request by nsrip-dd**
### What does this PR do?

Adds a `go_execution_trace_enabled` tag to uploaded profiles.

### Motivation

Indicate whether execution tracing is enabled, to distinguish between missing a
trace because we don't collect them every profiling cycle from missing a trace
because the feature isn't turned on.

While we're here, add these extra tracing tags to the batch object
that's built during the profiling cycle. We can't safely read the
execution trace config during the uploading, because it can be modified
concurrently by the profiler after https://github.com/DataDog/dd-trace-go/pull/1978. It makes more sense to have
the profiler add all the needed information, which it can do safely, so
that the upload step can just deal with uploading.

### Describe how to test/QA your changes

This PR includes a unit test.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality.
- [x] If this interacts with the agent in a new way, a system test has been added.

### CODE REVIEW

**Comment by nsrip-dd** [comment]
Moved back to a draft, I might change this to put the indicator in the event.json rather than as a tag per Ivo's internal RFC. Will resolve soon.

File: profiler/profiler.go [hunk 1]
```diff
@@ -259,6 +259,13 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 			seq:   p.seq,
 			host:  p.cfg.hostname,
 			start: now(),
+			extraTags: []string{
+				// go_execution_trace_enabled indicates whether execution
+				// tracing is enabled, to distinguish between missing a trace
+				// because we don't collect them every profiling cycle from
+				// missing a trace because the feature isn't turned on.
+				fmt.Sprintf("go_execution_trace_enabled:%v", p.cfg.traceConfig.Enabled),
```
Comment (felixge) [inline profiler/profiler.go:267]: Do we expect users to use this tag? If not, maybe we should use `_dd.profiler.go_execution_trace_enabled` ? This was proposed in Ivo's internal doc. A `_dd.profiler.` prefix would also make it easier to hide these kind of tags in the UI (if we want to hide them).

(This is just a question, I'm not asking you to change the tag right away)

**Review by felixge — APPROVED** [review]
Code LGTM. But let's try to discuss the question I added.

**Review by felixge — APPROVED** [review]
LGTM for the next release. That being said, I think we should do some more work on internal alignment on how we want to handle this consistently across client libs. Ivo's proposal was different.

**Pull request merged**